### ISSUE TITLE
Training does not converge

### ISSUE THREAD

**Issue by caxelos** [FrancescoSaverioZuppichini/ResNet#1]
* Thank you for your nice implementation of Resnets.
* I have a problem in **customizing** this implementation in my project,because outputs are taking **NaN** values.
* Other models that I have tested **work fine**,so I believe that the rest of code is fine.
* I pass to my resnet model **batches** with size **32**.Each batch contains images with size **60x36** pixels.I only changed the decoder,because I have 2 outputs and my problem is regression(514 features is the encoder's output,before applying the Linear layer).
* My outputs should be at range **[0,1]**.However, they get values:
- 1st iter:  output ε [0,1]
- 2nd iter: output ε [5,10]
- 3rd iter:  output ε [80,100]
- 4th iter:   output > 10000
- 5th iter:   output = Nan
 * Can this be a **gradient explosion**? Have you faced a similar with that  problem? The other models that I used **didn't** explode the gradients and that makes me curious.I share here my code,in case you want to take a look: https://github.com/trakaros/MPIIGaze/blob/hpc-code/python/pytorch/models/myresnet_basic.py

* Thanks in advance

**Comment by caxelos** [comment]
* Ok,I found out the solution.I should have chosen the correct number of layers,in order not to have problem with dimensions(which caused the warning).
* Reducing the num of layers from 4 to 2 and choosing the BasicBlock fixed my problem

**Comment by FrancescoSaverioZuppichini** [comment]
Yes, your images are too small :)
